src/features/device-heartbeat/index.ts

```diff
@@ -429,9 +430,21 @@ export class DeviceOnlineStateManager extends events.EventEmitter {
 		this.isConsuming = true;
 		this.consume();
 
-		deviceStateEvents.on('get-state', async (uuid) => {
+		deviceStateEvents.on('get-state', async (uuid, { apiKey }) => {
 			try {
-				const pollInterval = await getPollInterval(uuid);
+				const key = apiKey?.key;
+				if (typeof key !== 'string') {
+					return;
+				}
+
+				const [isDeviceApiKey, pollInterval] = await Promise.all([
+					isApiKeyOfActorType('device', key),
```

Comment:
I would rather this checks for the api key role via `device-api-key` because in theory we could have other device keys that are scoped/etc and should not update the status

Reply:
@Page- does `isApiKeyWithRole` sounds closed to what you mean?
If so, does it make sense to have it in `infra`?